pkg/executor/test/autoidtest/autoid_test.go

```diff
@@ -567,3 +567,43 @@ func TestIssue39528(t *testing.T) {
 	// Make sure the code does not visit tikv on allocate path.
 	require.False(t, codeRun)
 }
+
+func TestIssue52622(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec(`set @@auto_increment_increment = 66;`)
+	tk.MustExec(`set @@auto_increment_offset = 9527;`)
+
+	tk.MustQuery(`select @@auto_increment_increment;`).Check(testkit.Rows("66"))
+	tk.MustQuery(`select @@auto_increment_offset;`).Check(testkit.Rows("9527"))
+
+	for i := 0; i < 2; i++ {
+		createTableSQL := "create table issue52622 (id int primary key auto_increment, k int)"
+		if i == 0 {
+			createTableSQL = createTableSQL + " AUTO_ID_CACHE 1"
+		}
+
+		tk.MustExec(createTableSQL)
+		tk.MustExec("insert into issue52622 (k) values (1),(2),(3);")
+		tk.MustQuery("select * from issue52622").Check(testkit.Rows("1 1", "67 2", "133 3"))
+		if i == 0 {
+			tk.MustQuery("show create table issue52622").CheckContain("134")
+		}
+		tk.MustExec("insert into issue52622 (k) values (4);")
+		tk.MustQuery("select * from issue52622").Check(testkit.Rows("1 1", "67 2", "133 3", "199 4"))
+
+		tk.MustExec("truncate table issue52622;")
+		tk.MustExec("insert into issue52622 (k) values (1)")
+		tk.MustExec("insert into issue52622 (k) values (2)")
+		tk.MustExec("insert into issue52622 (k) values (3)")
+		if i == 0 {
+			tk.MustQuery("show create table issue52622").CheckContain("134")
+		}
+		tk.MustExec("insert into issue52622 (k) values (4);")
+		tk.MustQuery("select * from issue52622").Check(testkit.Rows("1 1", "67 2", "133 3", "199 4"))
```

Comment:
```
mysql> show variables like '%auto_increment%';
+--------------------------+-------+
| Variable_name            | Value |
+--------------------------+-------+
| auto_increment_increment | 66    |
| auto_increment_offset    | 9527  |
+--------------------------+-------+
2 rows in set (0.01 sec)

mysql> create table t (id int primary key auto_increment, k int);
ERROR 1046 (3D000): No database selected
mysql> use test;
Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

Database changed
mysql> create table t (id int primary key auto_increment, k int);
Query OK, 0 rows affected (0.01 sec)

mysql>
mysql> insert into t (k) values (1),(2),(3);
Query OK, 3 rows affected (0.00 sec)
Records: 3  Duplicates: 0  Warnings: 0

mysql> select * from t;
+-----+------+
| id  | k    |
+-----+------+
|   7 |    1 |
|  73 |    2 |
| 139 |    3 |
+-----+------+
3 rows in set (0.00 sec)
```

Reply:
I don't know why the result in my MySQL is different.

Reply:
oops, you're right, let me check again!

Reply:
Well, I find MySQL document and actual implementation mismatch.
In MySQL:

```
@@auto_increment_increment = 66 ,  @@auto_increment_offset = 9527　
insert into t values (1),(2),(3) get  7 1,  73 2,  139 3


 @@auto_increment_increment = 66 ,  @@auto_increment_offset = 1　
insert into t values (1),(2),(3) get    1 1,  67 2,  133 3
```

It's quite difficult to understand why.
So I decide to keep our implement following the MySQL document way, and won't change here.

@Defined2014

Reply:
Please fix the test.

Reply:
Done